modules/payment/tests/src/Functional/ManualPaymentAdminTest.php

```diff
@@ -0,0 +1,258 @@
+   * @var \Drupal\commerce_payment\Entity\PaymentGatewayInterface
+   */
+  protected $paymentGateway;
+
+  /**
+   * Admin's payment method.
+   *
+   * @var \Drupal\commerce_payment\Entity\PaymentMethodInterface
+   */
+  protected $paymentMethod;
+
+  /**
+   * The base admin payment uri.
+   *
+   * @var string
+   */
+  protected $paymentUri;
+
+  /**
+   * The admin's order.
+   *
+   * @var \Drupal\commerce_order\Entity\OrderInterface
+   */
+  protected $order;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static $modules = [
+    'commerce_order',
+    'commerce_product',
+    'commerce_payment',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getAdministratorPermissions() {
+    return array_merge([
+      'administer commerce_order',
+      'administer commerce_payment_gateway',
+      'administer commerce_payment',
+    ], parent::getAdministratorPermissions());
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    $profile = $this->createEntity('profile', [
+      'type' => 'customer',
+      'address' => [
+        'country_code' => 'US',
+        'postal_code' => '53177',
+        'locality' => 'Milwaukee',
+        'address_line1' => 'Pabst Blue Ribbon Dr',
+        'administrative_area' => 'WI',
+        'given_name' => 'Frederick',
+        'family_name' => 'Pabst',
+      ],
+      'uid' => $this->loggedInUser->id(),
+    ]);
+
+    $this->paymentGateway = $this->createEntity('commerce_payment_gateway', [
+      'id' => 'manual',
+      'label' => 'Manual example',
+      'plugin' => 'manual',
+    ]);
+    $this->paymentGateway->getPlugin()->setConfiguration([
+      'reusable' => '1',
+      'expires' => '',
+      'instructions' => [
+        'value' => 'Test instructions.',
+        'format' => 'plain_text',
+      ],
+    ]);
+    $this->paymentGateway->save();
+    $this->paymentMethod = $this->createEntity('commerce_payment_method', [
+      'uid' => $this->loggedInUser->id(),
+      'type' => 'manual',
+      'payment_gateway' => 'manual',
+      'billing_profile' => $profile,
+      'expires' => 0,
+    ]);
+
+    $details = ['manual' => ''];
+    $this->paymentGateway->getPlugin()->createPaymentMethod($this->paymentMethod, $details);
+
+    $variation = $this->createEntity('commerce_product_variation', [
+      'type' => 'default',
+      'sku' => 'test-product-01',
+      'price' => new Price('10', 'USD'),
+    ]);
+
+    $order_item = $this->createEntity('commerce_order_item', [
+      'type' => 'default',
+      'quantity' => 1,
+      'purchased_entity' => $variation,
+      'unit_price' => new Price('10', 'USD'),
+    ]);
+
+    $this->order = $this->createEntity('commerce_order', [
+      'uid' => $this->loggedInUser->id(),
+      'type' => 'default',
+      'state' => 'draft',
+      'order_items' => [$order_item],
+      'store_id' => $this->store,
+    ]);
+
+    $this->paymentUri = 'admin/commerce/orders/' . $this->order->id() . '/payments';
```

Comment:
We should generate this via `Url::fromRoute` with proper parameters. I know `drupalGet` tries, but we should use proper generation.